File: utils-io/src/main/java/org/wdt/utils/io/PathUtils.java
package org.wdt.utils.io;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.attribute.FileTime;
import java.time.Instant;
import java.util.List;

public class PathUtils {
    public static String readFileToString(Path FilePath) throws IOException {
        return IOUtils.toString(Files.newBufferedReader(FilePath));
    }

    public static void writeStringToFile(Path path, String s) throws IOException {
        touch(path);
        IOUtils.write(Files.newBufferedWriter(path), s);
    }

    public static void touch(Path path) throws IOException {
        if (!Files.exists(path)) {
            Files.createDirectories(path.getParent());
            Files.createFile(path);
        } else {
            Files.setLastModifiedTime(path, FileTime.from(Instant.now()));
        }
    }

    public static List<String> readFileToLines(Path path) throws IOException {
        return IOUtils.readLines(Files.newBufferedReader(path));
    }

    public static void delete(Path path) throws IOException {
        if (Files.exists(path)) {
            Files.delete(path);
        }
    }

    public static long sizeOf(Path path) throws IOException {
        return Files.size(path);
    }

    public static void cleanDirectory(Path path) throws IOException {
        ckeckIsDirectory(path);
        for (Path path1 : Files.newDirectoryStream(path)) {
            if (Files.isDirectory(path1)) {
                cleanDirectory(path1);
                deleteDirectory(path1);
            } else {
                delete(path1);
            }
        }
    }

    public static void deleteDirectory(Path path) throws IOException {
        ckeckIsDirectory(path);
        cleanDirectory(path);
        delete(path);
    }


    public static long sizeOfDirectory(Path path) throws IOException {
        ckeckIsDirectory(path);
        long dirSize = 0;
        for (Path path1 : Files.list(path).toList()) {
            if (Files.isDirectory(path1)) {
                sizeOfDirectory(path1);
            } else {
                dirSize = PathUtils.sizeOf(path1);
            }
        }
        return dirSize;
    }

    public static boolean isFileExists(Path path) {
        return Files.exists(path);
    }

    public static boolean isFileNotExists(Path path) {
        return Files.notExists(path);
    }

    public static boolean isFileNotExistsAndIsNotSameSize(Path path, long l) throws IOException {
        return isFileNotExists(path) && sizeOf(path) != l;
    }

    public static void ckeckIsDirectory(Path path) throws IOException {
        if (!Files.isDirectory(path)) {
            throw new IOException("This File isn't Directory");
        }
    }

    public static String getFileSha1(Path path) throws IOException {
        return IOUtils.getInputStreamSha1(Files.newInputStream(path));
    }

}

File: utils-io/src/main/kotlin/org/wdt/utils/io/PathUtils.kt
package org.wdt.utils.io

import java.nio.file.Path

fun Path.readFileToString(): String {
    return PathUtils.readFileToString(this)
}

fun Path.writeStringToFile(string: String) {
    PathUtils.writeStringToFile(this, string)
}

fun Path.writeStringToFile(string: String, boolean: Boolean) {
    PathUtils.writeStringToFile(this, string, boolean)
}

fun Path.touch() {
    PathUtils.touch(this)
}

fun Path.readFileToLines(): List<String> {
    return PathUtils.readFileToLines(this)
}

fun Path.delete() {
    PathUtils.delete(this)
}

fun Path.sizeOf(): Long {
    return PathUtils.sizeOf(this)
}

fun Path.cleanDirectory() {
    PathUtils.cleanDirectory(this)
}

fun Path.deleteDirectory() {
    PathUtils.deleteDirectory(this)
}

fun Path.sizeOfDirectory(): Long {
    return PathUtils.sizeOfDirectory(this)
}

fun Path.isFileExists(): Boolean {
    return PathUtils.isFileExists(this)
}

fun Path.isFileNotExistsAndIsNotSameSize(long: Long): Boolean {
    return PathUtils.isFileNotExistsAndIsNotSameSize(this, long)
}

fun Path.createDirectories() {
    PathUtils.createDirectories(this)
}

fun Path.getExtension(): String {
	return FilenameUtils.getExtension(this.toFile().canonicalPath)
}
fun Path.getFileSha1(): String {
    return PathUtils.getFileSha1(this)
}